sonic_y_cable/y_cable.py

```diff
@@ -507,3 +546,655 @@ def check_if_link_is_active_for_torB(physical_port):
+    been done manually by the user.
+
+    Args:
+        physical_port:
+             an Integer, the actual physical port connected to a Y cable
+    Returns:
+        an integer, the number of times manually the Y-cable has been switched
+    """
+
+    curr_offset = Y_CABLE_MANUAL_SWITCH_COUNT
+
+    count = 0
+
+    if platform_chassis is not None:
+        msb_result = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset, 1)
+        y_cable_validate_read_data(msb_result, 1, physical_port, "manual switch count msb result")
+        msb_result_1 = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset + 1, 1)
+        y_cable_validate_read_data(msb_result_1, 1, physical_port, "manual switch count msb result 1")
+        msb_result_2 = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset + 2, 1)
+        y_cable_validate_read_data(msb_result_2, 1, physical_port, "manual switch count msb result 2")
+        lsb_result = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset+3, 1)
+        y_cable_validate_read_data(lsb_result, 1, physical_port, "manual switch count lsb result")
+        count = (msb_result[0] << 24 | msb_result_1[0] << 16 | msb_result_2[0] << 8 | lsb_result[0])
+
+    else:
+        helper_logger.log_error("platform_chassis is not loaded, failed to get manual switch count")
+        return -1
+
+    return count
+
+
+@hook_y_cable_simulator
+def get_automatic_switch_count(physical_port):
+    """
+    This API specifically returns the switch count to change the Active TOR which has
+    been done automatically by the cable.
+
+    Args:
+        physical_port:
+             an Integer, the actual physical port connected to a Y cable
+    Returns:
+        an integer, the number of times automatically  the Y-cable has been switched
+    """
+
+    curr_offset = Y_CABLE_AUTO_SWITCH_COUNT
+
+    count = 0
+
+    if platform_chassis is not None:
+        msb_result = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset, 1)
+        y_cable_validate_read_data(msb_result, 1, physical_port, "auto switch count msb result")
+        msb_result_1 = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset + 1, 1)
+        y_cable_validate_read_data(msb_result_1, 1, physical_port, "auto switch count msb result 1")
+        msb_result_2 = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset + 2, 1)
+        y_cable_validate_read_data(msb_result_2, 1, physical_port, "auto switch count msb result 2")
+        lsb_result = platform_chassis.get_sfp(physical_port).read_eeprom(curr_offset+3, 1)
+        y_cable_validate_read_data(lsb_result, 1, physical_port, "auto switch count lsb result")
+        count = (msb_result[0] << 24 | msb_result_1[0] << 16 | msb_result_2[0] << 8 | lsb_result[0])
+
+    else:
+        helper_logger.log_error("platform_chassis is not loaded, failed to get manual switch count")
+        return -1
```

Comment:
This code is duplicated for manual and automatic switch count. Suggest creating a helper function which takes the starting offset as a parameter.

Reply:
fixed

Reply:
I don't see any change. Did a commit get lost?

Reply:
fixed, removed the duplicate code.